include/zephyr/net/net_if.h

```diff
@@ -125,6 +125,7 @@ struct net_if_addr {
 			uint8_t acd_state;
 		};
 #endif /* CONFIG_NET_IPV4_ACD */
+		uint8_t _dummy;
```

Comment:
It would be great to see some explanation in commit message why this is causing a difference in C vs C++. Is it because if DAD and ACD is not enabled, the union is empty which then causes the difference in C vs C++?